lookout/style/format/feature_extractor.py

```diff
@@ -796,15 +795,130 @@ def _compute_labels_mappings(self, vnodes: Iterable[VirtualNode]) -> None:
+    finished.
+
+    :param file: file annotated with `UASTAnnotation`, `PathAnnotation` and `RawTokenAnnotation`. \
+                 It is expected to be the output of  `FeatureExtractor._parse_file()`.
+    :return: The old `FeatureExtractor._parse_file()` output format, that is \
+             Tuple with `VirtualNode`-s and `bbfsh.Node` id to parent mapping.
+    """
+    vnodes = []
+    path = file.get(PathAnnotation).path
+    raw_lines_data = file.sequence.splitlines(keepends=True)
+    line_lens = [0] + [len(d) for d in raw_lines_data]
+    line_lens[-1] += 1
+    _line_start_offsets = numpy.array(line_lens).cumsum()
+    for annotation in file.iter_annotation(RawTokenAnnotation):
+        vnode = VirtualNode(
+            file[annotation.span],
+            _to_position(raw_lines_data, _line_start_offsets, annotation.start),
+            _to_position(raw_lines_data, _line_start_offsets, annotation.stop),
+            is_accumulated_indentation=False,
+            path=path,
+            node=annotation.node,
+            y=None,
+        )
+        vnodes.append(vnode)
+    return vnodes, file.get(UASTAnnotation).parents
+
+
+def file_to_vnodes_and_parents(file: AnnotationManager) -> Tuple[List["VirtualNode"],
+                                                                 Dict[int, bblfsh.Node]]:
+    """
+    Convert `AnnotationManager` instance to an old format, that is a sequence of vnodes and \
+    vnodes parents mapping.
+
+    The function is created for backwards capability and should be removed after refactoring is \
+    finished.
+
+    :param file: file annotated with `Path`-, `Token`-, `Label`-, `TokenParent`- `Annotation`.
+    :return: Tuple with `VirtualNode`-s and node id to parents mapping.
+    """
+    vnodes = []
+    path = file.get(PathAnnotation).path
+    raw_lines_data = file.sequence.splitlines(keepends=True)
+    line_lens = [0] + [len(d) for d in raw_lines_data]
+    line_lens[-1] += 1
+    _line_start_offsets = numpy.array(line_lens).cumsum()
+    vnode_parents = {}
+    for annotations in file.iter_annotations(TokenAnnotation,
+                                             LabelAnnotation, TokenParentAnnotation):
+        vnode = VirtualNode(
+            file[annotations.span],
+            _to_position(raw_lines_data, _line_start_offsets, annotations.start),
+            _to_position(raw_lines_data, _line_start_offsets, annotations.stop),
+            is_accumulated_indentation=False,
+            path=path,
+            node=annotations[TokenAnnotation].node,
+            y=annotations[LabelAnnotation].label if LabelAnnotation in annotations else None,
+        )
+        vnodes.append(vnode)
+        vnode_parents[id(vnode)] = annotations[TokenParentAnnotation].parent \
+            if TokenParentAnnotation in annotations else None
+    return vnodes, vnode_parents
+
+
+def to_vnodes_format(files: Sequence[AnnotationManager],
```

Comment:
Here comes the third function which is named very similar to the other two and has another obvious docstring :) How is it different from

```python
for file in files:
    results.append(file_to_vnodes_and_parents(file))
```

? Why do we need it? (all those are the issues with naming and docstrings, not with the logic)

Reply:
explained in docstring.